doc/misc/aliases.7.rst

```diff
@@ -0,0 +1,261 @@
+    - ``group_id`` - a group this alias is part of if any
+    - ``complete`` - whether bash autocompletion should be used for this alias
+    - ``attached_named_args`` - options that will be used with the command. The format is an array of inline
+      tables, each of which must contain an ``id_path`` key to specify the path to an option, and may contain also
+      a ``value`` key to specify the value of the option.
+
+The required keys are ``type``, and ``attached_command``.
+
+Examples:
+  - Alias ``grouplist`` for ``group list``:
+
+    .. code-block:: none
+
+        ['grouplist']
+        type = 'command'
+        attached_command = 'group.list'
+        descr = "Alias for 'group list'"
+
+  - Alias ``group.ls`` for ``group list``:
+
+    .. code-block:: none
+
+        ['group.ls']
+        type = 'command'
+        attached_command = 'group.list'
+        descr = "Alias for 'group list'"
+        complete = true
+
+  - Alias ``list-fedora-all`` for ``--repo=fedora list --showduplicates``:
+
+    .. code-block:: none
+
+        ['list-fedora-all']
+        type = 'command'
+        attached_command = 'list'
+        descr = "Alias for '--repo=fedora list --showduplicates'"
+        complete = true
+        attached_named_args = [
+            { id_path = 'repo', value = 'fedora' },
+            { id_path = 'list.showduplicates' }
+        ]
+
+
+.. _aliases_misc_cloned_named_arg_ref-label:
+
+Type: cloned_named_arg
+----------------------
+
+The ``cloned_named_arg`` alias defines another name for a given option.
+
+Keys:
+    - ``type`` with value ``cloned_named_arg``
+    - ``long_name`` - name of the alias option
+    - ``short_name`` - one-letter shortcut of the name
+    - ``source`` - path to the option for which this alias is defined
+    - ``group_id`` - a group this alias is part of if any
+    - ``complete`` - whether bash autocompletion should be used for this alias
+
+The required keys are ``type``, either ``long_name`` or ``short_name``, and ``source``.
+
+Examples:
+  - Alias ``--nobest`` for ``--no-best``:
+
+    .. code-block:: none
+
+        ['nobest']
+        type = 'cloned_named_arg'
+        long_name = 'nobest'
+        source = 'no-best'
+
+  - Alias ``repoquery --list`` or ``repoquery -l`` for ``repoquery --files``:
+
+    .. code-block:: none
+
+        ['repoquery.list']
+        type = 'cloned_named_arg'
+        long_name = 'list'
+        short_name = 'l'
+        source = 'repoquery.files'
+
+
+.. _aliases_misc_named_arg_ref-label:
+
+Type: named_arg
+---------------
+
+The ``named_arg`` defines an alias that can replace multiple options and can define a value for each.
+
+Keys:
+    - ``type`` with value ``named_arg``
+    - ``long_name`` - name of the alias option
+    - ``short_name`` - one-letter shortcut of the name
+    - ``descr`` - description that will be shown in help
+    - ``has_value`` - whether the option requires a value. The value is then substituted for ``${}`` strings in the
```

Comment:
Just a suggestion but perhaps we could specify that both `has_value` and `complete` are set to `false` by default?